flang/lib/Lower/Allocatable.cpp

```diff
@@ -269,19 +299,51 @@ class AllocateStmtHelper {
     }
   }
 
+  /// Lower the length parameters that may be specified in the optional
+  /// type specification.
+  void lowerAllocateLengthParameters() {
+    const auto *typeSpec = getAllocateStmtTypeSpec();
+    if (!typeSpec)
```

Comment:
Hmm. Do deferred / non-deferred length type parameters have something to do with the shape specification of an allocate-stmt? (They shouldn't, as I believe they are distinct semantically.)

Reply:
No, they do not impact the shape, but they impact the type-spec that may be provided or not in an allocate statement.